### PR TITLE
Fix #268: add printer_options configuration key

### LINKED ISSUE #268: Q: How to configure pibooth to print borderless with Canon Selphy CP1300?
The problem is that I have a small (white) border on the printed image but if I use `lp 2021-07-25-09-47-26_pibooth.jpg` the image is printed borderless like expected. Any help would be appreciated and this app is really cool.

How can I configure `pibooth` to use the default print settings of the system. The [documentation](https://pibooth.readthedocs.io/en/latest/sources/start.html?highlight=paper#printer) states "Here is the default configuration used for this project in CUPS, it may depend on the printer used" but I cannot find any configuration option to change it or to use the system printer settings.

Should an option like `{'fit-to-page':'True'}` added to the [`printFile`](https://github.com/pibooth/pibooth/blob/0096604b09d65f7cae3880d78a0bb14defff27d0/pibooth/printer.py#L106) definition? Best way would be to use the printer system settings or make it configurable to use system printer default settings.

### Sytem printer configuration

Here is how I configured the Canon Selphy CP1300 printer to print borderless for the postcard format (just for reference /cc @mattdsteele).

```
lpinfo -m | grep cp1300
lpadmin -p pibooth -E -v gutenprint53+usb://canon-cp1300/B721010709211348 -m gutenprint.5.3://canon-cp1300/expert
lpoptions -d pibooth
lpstat -s
lpstat -p pibooth
lpadmin -p pibooth -o printer-error-policy=abort-job

lpoptions -p pibooth -o StpBorderless=True
lpoptions -p pibooth -o StpiShrinkOutput=Expand
lpoptions -p pibooth -o StpColorPrecision=Best
lpoptions -p pibooth -o StpImageType=Photo

cupsenable pibooth
cupsaccept pibooth
```
The command `lpoptions -p pibooth -l` shows these default settings:

```
ageSize/Media Size: *Postcard w253h337 w155h244
ColorModel/Color Model: Gray Black *RGB CMY CMYK KCMY
StpColorPrecision/Color Precision: Normal *Best
Resolution/Resolution: *300dpi
StpiShrinkOutput/Shrink Page If Necessary to Fit Borders: Shrink Crop *Expand
StpBorderless/Borderless: False *True
StpColorCorrection/Color Correction: *None Accurate Bright Hue Uncorrected Desaturated Threshold Density Raw Predithered
StpBrightness/Brightness: 0 100 200 300 400 500 600 700 800 900 *None 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000 Custom.REAL
StpFineBrightness/Brightness Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpContrast/Contrast: 0 100 200 300 400 500 600 700 800 900 *None 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000 2100 2200 2300 2400 2500 2600 2700 2800 2900 3000 3100 3200 3300 3400 3500 3600 3700 3800 3900 4000 Custom.REAL
StpFineContrast/Contrast Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpSaturation/Saturation: 0 100 200 300 400 500 600 700 800 900 *None 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000 2100 2200 2300 2400 2500 2600 2700 2800 2900 3000 3100 3200 3300 3400 3500 3600 3700 3800 3900 4000 4100 4200 4300 4400 4500 4600 4700 4800 4900 5000 5100 5200 5300 5400 5500 5600 5700 5800 5900 6000 6100 6200 6300 6400 6500 6600 6700 6800 6900 7000 7100 7200 7300 7400 7500 7600 7700 7800 7900 8000 8100 8200 8300 8400 8500 8600 8700 8800 8900 9000 Custom.REAL
StpFineSaturation/Saturation Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpImageType/Image Type: None Text Graphics TextGraphics *Photo LineArt
StpGamma/Composite Gamma: 100 200 300 400 500 600 700 800 900 1000 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000 2100 2200 2300 2400 2500 2600 2700 2800 2900 3000 3100 3200 3300 3400 3500 3600 3700 3800 3900 4000 *None Custom.REAL
StpFineGamma/Composite Gamma Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpCyanGamma/Cyan: 0 100 200 300 400 500 600 700 800 900 1000 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000 2100 2200 2300 2400 2500 2600 2700 2800 2900 3000 3100 3200 3300 3400 3500 3600 3700 3800 3900 4000 *None Custom.REAL
StpFineCyanGamma/Cyan Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpMagentaGamma/Magenta: 0 100 200 300 400 500 600 700 800 900 1000 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000 2100 2200 2300 2400 2500 2600 2700 2800 2900 3000 3100 3200 3300 3400 3500 3600 3700 3800 3900 4000 *None Custom.REAL
StpFineMagentaGamma/Magenta Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpYellowGamma/Yellow: 0 100 200 300 400 500 600 700 800 900 1000 1100 1200 1300 1400 1500 1600 1700 1800 1900 2000 2100 2200 2300 2400 2500 2600 2700 2800 2900 3000 3100 3200 3300 3400 3500 3600 3700 3800 3900 4000 *None Custom.REAL
StpFineYellowGamma/Yellow Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpCyanBalance/Cyan Balance: 0 100 200 300 400 500 600 700 800 900 1000 *None Custom.REAL
StpFineCyanBalance/Cyan Balance Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpMagentaBalance/Magenta Balance: 0 100 200 300 400 500 600 700 800 900 1000 *None Custom.REAL
StpFineMagentaBalance/Magenta Balance Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpYellowBalance/Yellow Balance: 0 100 200 300 400 500 600 700 800 900 1000 *None Custom.REAL
StpFineYellowBalance/Yellow Balance Fine Adjustment: *None 0 5 10 15 20 25 30 35 40 45 50 55 60 65 70 75 80 85 90 95
StpLinearContrast/Linear Contrast Adjustment: *False True
```

### Config
This is my `pibooth v2.0.3` configuration:

```
[GENERAL]
# User interface language: 'de', 'dk', 'en', 'es', 'fr', 'hu', 'nl' or 'no'
language = de 

# Path to save pictures (list of quoted paths accepted)
directory = ~/Pictures/pibooth

# Start pibooth at Raspberry Pi startup
autostart = False 

# In debug mode, exceptions are not caught, logs are more verbose, pictures are cleared at startup
debug = False 

# Path to custom plugin(s) not installed with pip (list of quoted paths accepted)
plugins = 

# Plugin names to be disabled after startup (list of quoted names accepted)
plugins_disabled = 

# Enable a virtual keyboard in the settings interface
vkeyboard = False

[WINDOW]
# The (width, height) of the display window or 'fullscreen'
size = (1280, 800)
#size=fullscreen

# Background RGB color or image path
background = (0, 0, 0)

# Text RGB color
text_color = (255, 255, 255)

# Blinking background when a capture is taken
flash = True

# Animate the last taken picture by displaying captures one by one
animate = False

# How long is displayed the capture in seconds before switching to the next one
animate_delay = 0.2

# On 'finish' state: how long is displayed the final picture in seconds (0 if never shown)
finish_image_delay = 0

# On 'wait' state: how long is displayed the final picture in seconds before being hidden (-1 if never hidden)
wait_image_delay = -1

# Show arrows to indicate physical buttons: 'bottom', 'top', 'hidden' or 'touchscreen'
#arrow = hidden
arrows = touchscreen 

# Apply horizontal offset to arrows position
arrows_x_offset = 0

# How long is the preview in seconds
preview_delay = 3

# Show a countdown timer during the preview
preview_countdown = True

# Stop the preview before taking the capture
preview_stop_on_capture = False

[PICTURE]
# Orientation of the final picture: 'auto', 'portrait' or 'landscape'
orientation = landscape 

# Possible choice(s) of captures numbers (numbers between 1 to 4)
captures = (4, 1)

# Effect applied to the captures (list of quoted names accepted)
captures_effects = none

# Crop each capture border in order to fit the paper size
captures_cropping = False
#captures_cropping = True 

# Thick (in pixels) between captures and picture borders/texts
#margin_thick = 100
margin_thick = 0

# Main text displayed
footer_text1 = 

# Secondary text displayed
footer_text2 = 

# RGB colors used for footer texts (list of tuples accepted)
text_colors = (0, 0, 0)

# Fonts name or file path used for footer texts (list of quoted names accepted)
text_fonts = ('Amatic-Bold', 'AmaticSC-Regular')

# Alignments used for footer texts: 'left', 'center' or 'right' (list of quoted names accepted)
text_alignments = center

# Overlay path (PNG file) with same aspect ratio than final picture (list of quoted paths accepted)
overlays = 

# Background RGB color or image path (list of tuples or quoted paths accepted)
backgrounds = (255, 255, 255)

[CAMERA]
# Adjust ISO for lighting issues, can be different for preview and capture (list of integer accepted)
iso = 100

# Flip horizontally the capture
flip = False

# Rotation of the camera: 0, 90, 180 or 270
rotation = 0

# Resolution for camera captures (preview will have same aspect ratio)
#resolution = (1934, 2464)
resolution = (5472, 3648)

# Delete captures from camera internal memory (when applicable)
delete_internal_memory = False

[PRINTER]
# Name of the printer defined in CUPS (or use the 'default' one)
printer_name = default

# How long is the print view in seconds (0 to skip it)
printer_delay = 10

# Maximum number of printed pages before warning on paper/ink levels (-1 = infinite)
max_pages = -1 

# Maximum number of duplicate pages sent to the printer (avoid paper waste)
max_duplicates = 1 

# Print 1, 2, 3 or 4 picture copies per page
pictures_per_page = 1

[CONTROLS]
# How long to press a single hardware button in seconds
debounce_delay = 0.3

# How long to press multiple hardware buttons in seconds
multi_press_delay = 0.5

# Physical GPIO IN pin to take a picture
picture_btn_pin = 11

# Physical GPIO OUT pin to light a LED when picture button is pressed
picture_led_pin = 7

# Physical GPIO IN pin to print a picture
print_btn_pin = 13

# Physical GPIO OUT pin to light a LED when print button is pressed
print_led_pin = 15
```

### PR DESCRIPTION
**Pull request by anxuae**

### CODE REVIEW

File: pibooth/printer.py [hunk 1]
```diff
@@ -101,9 +108,9 @@ def print_file(self, filename, copies=1):
                 # are the one necessary to render several pictures on same page.
                 factory.set_margin(2)
                 factory.save(fp.name)
-                self._conn.printFile(self.name, fp.name, osp.basename(filename), {})
+                self._conn.printFile(self.name, fp.name, osp.basename(filename), self.options)
         else:
-            self._conn.printFile(self.name, filename, osp.basename(filename), {})
+            self._conn.printFile(self.name, filename, osp.basename(filename), self.options)
         LOGGER.debug("File '%s' sent to the printer", filename)
```
Comment (sandrokeil) [inline pibooth/printer.py:114]: @anxuae Can we add the print options to the log output too e.g. `LOGGER.debug("File '%s' sent to the printer with options %s", filename, str(self.options))`

**Comment by sandrokeil** [comment]
I can confirm hat it works.

```
$ pibooth-printcfg 
pygame 2.0.1 (SDL 2.0.9, Python 3.7.3)
Hello from the pygame community. https://www.pygame.org/contribute.html
pygame-menu 4.0.6
INFO:pibooth:Connected to printer 'pibooth'
PageSize = Postcard
     Media Size
     Choices: Postcard = Postcard 100x148mm
              w253h337 = CP_L 89x119mm
              w155h244 = Card 54x86mm

PageRegion = Postcard
     Media Size
     Choices: Postcard = Postcard 100x148mm
              w253h337 = CP_L 89x119mm
              w155h244 = Card 54x86mm

ColorModel = RGB
     Color Model
     Choices: Gray = Grayscale
              Black = Inverted Grayscale
              RGB = RGB Color
              CMY = CMY Color
              CMYK = CMYK
              KCMY = KCMY

StpColorPrecision = Best
     Color Precision
     Choices: Normal, Best

Resolution = 300dpi
     Resolution
     Choices: 300dpi = 300x300 DPI

StpiShrinkOutput = Expand
     Shrink Page If Necessary to Fit Borders
     Choices: Shrink = Shrink (print the whole page)
              Crop = Crop (preserve dimensions)
              Expand = Expand (use maximum page area)

StpBorderless = False
     Borderless
     Choices: False = No
              True = Yes

StpColorCorrection = None
     Color Correction
     Choices: None = Default
              Accurate = High Accuracy
              Bright = Bright Colors
              Hue = Correct Hue Only
              Uncorrected = Uncorrected
              Desaturated = Desaturated
              Threshold = Threshold
              Density = Density
              Raw = Raw
              Predithered = Pre-Dithered

StpBrightness = None
     Brightness
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              None = 1.000
              1100 = 1.100
              1200 = 1.200
              1300 = 1.300
              1400 = 1.400
              1500 = 1.500
              1600 = 1.600
              1700 = 1.700
              1800 = 1.800
              1900 = 1.900
              2000 = 2.000
              Custom = Custom

StpFineBrightness = None
     Brightness Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpContrast = None
     Contrast
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              None = 1.000
              1100 = 1.100
              1200 = 1.200
              1300 = 1.300
              1400 = 1.400
              1500 = 1.500
              1600 = 1.600
              1700 = 1.700
              1800 = 1.800
              1900 = 1.900
              2000 = 2.000
              2100 = 2.100
              2200 = 2.200
              2300 = 2.300
              2400 = 2.400
              2500 = 2.500
              2600 = 2.600
              2700 = 2.700
              2800 = 2.800
              2900 = 2.900
              3000 = 3.000
              3100 = 3.100
              3200 = 3.200
              3300 = 3.300
              3400 = 3.400
              3500 = 3.500
              3600 = 3.600
              3700 = 3.700
              3800 = 3.800
              3900 = 3.900
              4000 = 4.000
              Custom = Custom

StpFineContrast = None
     Contrast Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpSaturation = None
     Saturation
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              None = 1.000
              1100 = 1.100
              1200 = 1.200
              1300 = 1.300
              1400 = 1.400
              1500 = 1.500
              1600 = 1.600
              1700 = 1.700
              1800 = 1.800
              1900 = 1.900
              2000 = 2.000
              2100 = 2.100
              2200 = 2.200
              2300 = 2.300
              2400 = 2.400
              2500 = 2.500
              2600 = 2.600
              2700 = 2.700
              2800 = 2.800
              2900 = 2.900
              3000 = 3.000
              3100 = 3.100
              3200 = 3.200
              3300 = 3.300
              3400 = 3.400
              3500 = 3.500
              3600 = 3.600
              3700 = 3.700
              3800 = 3.800
              3900 = 3.900
              4000 = 4.000
              4100 = 4.100
              4200 = 4.200
              4300 = 4.300
              4400 = 4.400
              4500 = 4.500
              4600 = 4.600
              4700 = 4.700
              4800 = 4.800
              4900 = 4.900
              5000 = 5.000
              5100 = 5.100
              5200 = 5.200
              5300 = 5.300
              5400 = 5.400
              5500 = 5.500
              5600 = 5.600
              5700 = 5.700
              5800 = 5.800
              5900 = 5.900
              6000 = 6.000
              6100 = 6.100
              6200 = 6.200
              6300 = 6.300
              6400 = 6.400
              6500 = 6.500
              6600 = 6.600
              6700 = 6.700
              6800 = 6.800
              6900 = 6.900
              7000 = 7.000
              7100 = 7.100
              7200 = 7.200
              7300 = 7.300
              7400 = 7.400
              7500 = 7.500
              7600 = 7.600
              7700 = 7.700
              7800 = 7.800
              7900 = 7.900
              8000 = 8.000
              8100 = 8.100
              8200 = 8.200
              8300 = 8.300
              8400 = 8.400
              8500 = 8.500
              8600 = 8.600
              8700 = 8.700
              8800 = 8.800
              8900 = 8.900
              9000 = 9.000
              Custom = Custom

StpFineSaturation = None
     Saturation Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpImageType = TextGraphics
     Image Type
     Choices: None = Manual Control
              Text = Text
              Graphics = Graphics
              TextGraphics = Mixed Text and Graphics
              Photo = Photograph
              LineArt = Line Art

StpGamma = None
     Composite Gamma
     Choices: 100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              1000 = 1.000
              1100 = 1.100
              1200 = 1.200
              1300 = 1.300
              1400 = 1.400
              1500 = 1.500
              1600 = 1.600
              1700 = 1.700
              1800 = 1.800
              1900 = 1.900
              2000 = 2.000
              2100 = 2.100
              2200 = 2.200
              2300 = 2.300
              2400 = 2.400
              2500 = 2.500
              2600 = 2.600
              2700 = 2.700
              2800 = 2.800
              2900 = 2.900
              3000 = 3.000
              3100 = 3.100
              3200 = 3.200
              3300 = 3.300
              3400 = 3.400
              3500 = 3.500
              3600 = 3.600
              3700 = 3.700
              3800 = 3.800
              3900 = 3.900
              4000 = 4.000
              None = None
              Custom = Custom

StpFineGamma = None
     Composite Gamma Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpCyanGamma = None
     Cyan
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              1000 = 1.000
              1100 = 1.100
              1200 = 1.200
              1300 = 1.300
              1400 = 1.400
              1500 = 1.500
              1600 = 1.600
              1700 = 1.700
              1800 = 1.800
              1900 = 1.900
              2000 = 2.000
              2100 = 2.100
              2200 = 2.200
              2300 = 2.300
              2400 = 2.400
              2500 = 2.500
              2600 = 2.600
              2700 = 2.700
              2800 = 2.800
              2900 = 2.900
              3000 = 3.000
              3100 = 3.100
              3200 = 3.200
              3300 = 3.300
              3400 = 3.400
              3500 = 3.500
              3600 = 3.600
              3700 = 3.700
              3800 = 3.800
              3900 = 3.900
              4000 = 4.000
              None = None
              Custom = Custom

StpFineCyanGamma = None
     Cyan Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpMagentaGamma = None
     Magenta
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              1000 = 1.000
              1100 = 1.100
              1200 = 1.200
              1300 = 1.300
              1400 = 1.400
              1500 = 1.500
              1600 = 1.600
              1700 = 1.700
              1800 = 1.800
              1900 = 1.900
              2000 = 2.000
              2100 = 2.100
              2200 = 2.200
              2300 = 2.300
              2400 = 2.400
              2500 = 2.500
              2600 = 2.600
              2700 = 2.700
              2800 = 2.800
              2900 = 2.900
              3000 = 3.000
              3100 = 3.100
              3200 = 3.200
              3300 = 3.300
              3400 = 3.400
              3500 = 3.500
              3600 = 3.600
              3700 = 3.700
              3800 = 3.800
              3900 = 3.900
              4000 = 4.000
              None = None
              Custom = Custom

StpFineMagentaGamma = None
     Magenta Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpYellowGamma = None
     Yellow
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              1000 = 1.000
              1100 = 1.100
              1200 = 1.200
              1300 = 1.300
              1400 = 1.400
              1500 = 1.500
              1600 = 1.600
              1700 = 1.700
              1800 = 1.800
              1900 = 1.900
              2000 = 2.000
              2100 = 2.100
              2200 = 2.200
              2300 = 2.300
              2400 = 2.400
              2500 = 2.500
              2600 = 2.600
              2700 = 2.700
              2800 = 2.800
              2900 = 2.900
              3000 = 3.000
              3100 = 3.100
              3200 = 3.200
              3300 = 3.300
              3400 = 3.400
              3500 = 3.500
              3600 = 3.600
              3700 = 3.700
              3800 = 3.800
              3900 = 3.900
              4000 = 4.000
              None = None
              Custom = Custom

StpFineYellowGamma = None
     Yellow Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpCyanBalance = None
     Cyan Balance
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              1000 = 1.000
              None = None
              Custom = Custom

StpFineCyanBalance = None
     Cyan Balance Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpMagentaBalance = None
     Magenta Balance
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              1000 = 1.000
              None = None
              Custom = Custom

StpFineMagentaBalance = None
     Magenta Balance Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpYellowBalance = None
     Yellow Balance
     Choices: 0 = 0.000
              100 = 0.100
              200 = 0.200
              300 = 0.300
              400 = 0.400
              500 = 0.500
              600 = 0.600
              700 = 0.700
              800 = 0.800
              900 = 0.900
              1000 = 1.000
              None = None
              Custom = Custom

StpFineYellowBalance = None
     Yellow Balance Fine Adjustment
     Choices: None = 0.000
              0 = 0.000
              5 = 0.005
              10 = 0.010
              15 = 0.015
              20 = 0.020
              25 = 0.025
              30 = 0.030
              35 = 0.035
              40 = 0.040
              45 = 0.045
              50 = 0.050
              55 = 0.055
              60 = 0.060
              65 = 0.065
              70 = 0.070
              75 = 0.075
              80 = 0.080
              85 = 0.085
              90 = 0.090
              95 = 0.095

StpLinearContrast = False
     Linear Contrast Adjustment
     Choices: False = No
              True = Yes
```

And I also added these config options:

```
printer_options = {'StpBorderless': 'True', 'StpImageType': 'Photo'}
```

File: pibooth/printer.py (hunk 1 above)
Comment (anxuae) [reply pibooth/printer.py]: added

**Comment by DukeMaxwell** [comment]
Sorry to ask so long after the inital conversation, but could you explain to a noob how to add all that stuff to the printcfg?
Mine looks like 



```
$ pibooth-printcfg
pygame 1.9.6
Hello from the pygame community. https://www.pygame.org/contribute.html
pygame-menu 4.0.7
INFO:pibooth:Connected to printer 'Canon_SELPHY_CP1300_USB_'
PageSize = Custom.100x148mm
     Description: Media Size
     Choices:     54x86mm = 54x86mm
                  89x119mm = 89x119mm
                  Postcard = Hagaki
                  Custom = Custom
                  Custom.100x148mm = Custom.100x148mm

PageRegion = Custom.100x148mm
     Description: Media Size
     Choices:     54x86mm = 54x86mm
                  89x119mm = 89x119mm
                  Postcard = Hagaki
                  Custom = Custom
                  Custom.100x148mm = Custom.100x148mm

ColorModel = RGB
     Description: Print Color Mode
     Choices:     Gray = Monochrome
                  RGB = Color

OutputBin = FaceUp
     Description: Output Tray
     Choices:     FaceUp = Face Up

cupsPrintQuality = Normal
     Description: Print Quality
     Choices:     Normal

print-content-optimize = photo
     Description: Print Optimization
     Choices:     photo = Photo
                  auto = Automatic

print-rendering-intent = relative
     Description: Print Rendering Intent
     Choices:     relative = Relative
                  relative-bpc = Relative w/Black Point Compensation

print-scaling = auto
     Description: Print Scaling
     Choices:     auto = Automatic
                  auto-fit = Auto-fit
                  fill = Fill
                  fit = Fit
                  none = None

**Comment by VincentAulnay** [comment]
@sandrokeil how do you do to set your printer config ?
like @DukeMaxwell , I have the conifg about a CP1500.

**Pull request merged**